src/sagemaker/fw_utils.py

```diff
@@ -354,7 +363,12 @@ def tar_and_upload_dir(
         else:
             extra_args = None
 
-        session.resource("s3").Object(bucket, key).upload_file(tar_file, ExtraArgs=extra_args)
+        if s3_resource is None:
+            s3_resource = session.resource("s3")
```

Comment:
I know this wasn't your code originally, but can you supply the `region_name` arg?